homework-4/src/components/features/shows/ShowDetailsSection/ShowDetailsSection.tsx

```diff
@@ -0,0 +1,32 @@
+'use client';
+import { useParams } from "next/navigation";
+import useSWR from "swr";
+import MainLayout from "@/components/shared/MainLayout/MainLayout";
+import LoadingSpinner from "@/components/core/LoadingSpinner/LoadingSpinner";
+import { swrKeys } from "@/fetchers/swrKeys";
+import { fetcher } from "@/fetchers/fetcher";
+import { IShow } from "@/typings/show";
+
+interface ApiResponse {
+  show: IShow;
+}
+
+export default function ShowDetailsPage() {
+  const { id } = useParams();
+  const { data, error, isLoading } = useSWR<ApiResponse>(swrKeys.getShow(id as string), fetcher);
+  const show = data?.show;
+  
+  if (error) {
+    return <div>No show available</div>;
+  }
+
+  if (isLoading || !data) {
+    return <LoadingSpinner />;
+  }
```

Comment:
Kad iz nekog razloga ne dobijemo podatke i dalje će se nastavit vrtit spinner, što je loš UX.
Trebalo bi returnati ili neku poruku ili empty state komponentu. Odnosno, treba li nam ovdje provjera za `!data`